model/version_db.go

```diff
@@ -97,20 +97,23 @@ func byLatestProjectVersion(projectId string) bson.M {
 	}
 }
 
-// FindLatestRevisionForProject returns the latest revision for the project, and returns an error if it's not found.
-func FindLatestRevisionForProject(projectId string) (string, error) {
+// FindLatestRevisionAndauthorForProject returns the latest revision for the project, and returns an error if it's not found.
+func FindLatestRevisionAndAuthorForProject(projectId string) (string, string, error) {
 	v, err := VersionFindOne(db.Query(byLatestProjectVersion(projectId)).
-		Sort([]string{"-" + VersionRevisionOrderNumberKey}).WithFields(VersionRevisionKey))
+		Sort([]string{"-" + VersionRevisionOrderNumberKey}).WithFields(VersionRevisionKey, VersionAuthorKey))
 	if err != nil {
-		return "", errors.Wrapf(err, "finding most recent version for project '%s'", projectId)
+		return "", "", errors.Wrapf(err, "finding most recent version for project '%s'", projectId)
 	}
 	if v == nil {
-		return "", errors.Errorf("no recent version found for project '%s'", projectId)
+		return "", "", errors.Errorf("no recent version found for project '%s'", projectId)
 	}
 	if v.Revision == "" {
-		return "", errors.Errorf("latest version '%s' has no revision", v.Id)
+		return "", "", errors.Errorf("latest version '%s' has no revision", v.Id)
 	}
-	return v.Revision, nil
+	if v.Author == "" {
+		return "", "", errors.Errorf("latest version '%s' has no author", v.Id)
```

Comment:
(starting a thread to respond to your PR comments, just for easier organizing)
I think not returning an error here for the next year is reasonable, since this isn't a likely state. Although I think what you proposed in the ticket as an option (using a generic author for periodic builds, such as https://github.com/evergreen-ci/evergreen/blob/ff588a8d3f78ef0f34285b1041f17740a21087df/globals.go#L18) would solve this as well, since j.addVersion would just use this author directly. 

I think this could also be the better user experience -- I think it could be confusing for users to be alerted on or asked questions about tasks that they didn't themselves choose to run.

Reply:
Also it was nice of you to take this ticket! But hoping you took it for interest and not because you felt you needed to -- project tickets don't show up in our prioritized on deck but just let us know if there's work you'd like us to prioritize for a project (or that we forgot about) and we'll make sure it gets assigned.

Reply:
A generic user could work, but is notably different from buildvariant cron tasks where the author seems to be set as the author of the latest version, even though the task was run on a schedule. I think it's a bit confusing to have different behavior for periodic builds and cron builds since the scheduling aspect of them is somewhat similar. But I can also see how it'd be nice, so if that's what you think we should do, I can quickly change this PR to:

* Not error out on a missing author
* Always (or maybe conditionally, if the author is missing?) set the author to a generic user

No worries on the ticket! I was more curious how this worked and realized I could at the very least get things rolling in case it was useful for someone else. It didn't seem like that challenging to fix other than the design aspect of it all!

Reply:
Yeah I think cron builds _are_ different because they manage the scheduling behavior of _mainline commits_ , which have an author associated with them because they were created after a commit was pushed by that author, whereas periodic builds are really created _by the project_. That said I'm fine with conditionally setting to a generic author. Overall I prefer this to erroring, since there are more edge cases we may see.

Reply:
OK, that makes sense to me. I can change to a conditional generic user -- any thoughts on what you'd like the name to be?

Reply:
So for a generic user I think we'll need to insert a service user into the DB -- "Evergreen Periodic Build" makes sense to me, maybe with periodic_build as the user ID? Once you know what you're thinking for it, I can create a new API user for it.

Reply:
I made roughly this change. Let me know what you think, and otherwise what you said sounds good to me! I didn't fetch the User ID (that's AuthorID in a Revision, right?), but if you think we should have all of that when creating the new version I can do that, too.

Reply:
Can you do this the way I linked above instead? (Adding ID to globals and then calling that variable)

Reply:
Done!

Reply:
For better organization, can you move this variable to where the other IDs are that I linked? And this should be defined the same way -- you're storing the _id_, not the display name.

Reminder as Kim mentioned the best way to ensure a timely review is to click re-request at the top!

Reply:
Sorry, maybe I am misunderstanding. A `model.Revision` has both an `Author` and `AuthorID` field. From what I can tell, it seems like the `Author` field is what the `ActivatedBy` gets sets to later on, but I could be wrong. Do you think *both* should be set? Do I need to alter the code to also fetch `AuthorID` from the database when creating a new Revision as well?

Reply:
Oh I see. I think passing the author through the revision is actually a bit hacky, since the "activator" isn't necessarily associated with the revision, and yes we should have the ID and email stored, not just the display name. (Even though all you need is the display name, we shouldn't do the thing halfway, and it's not good practice to be hardcoding a display name as a const; we should only store IDs since these are immutable). 

I would recommend instead returning the authorId from the version, looking up the version, and storing that with VersionMetadata instead of in the revision. This will populate the fields you want at the bottom.

https://github.com/evergreen-ci/evergreen/blob/d6297fda59b13b9257ac912f614b8f94c2053798/repotracker/repotracker.go#L755

Reply:
(I would say I understand the change well enough to take over the PR if you would preer; not sure if you're still enjoying this haha)

Reply:
Ah, indeed this became a bit more complex than I thought! I'd be quite happy to have you take over at this point. Feel free to take any or none of the code from this PR!